### PR TITLE
Added Openstack security group caching.

### PR DESCRIPTION
**Pull request by varikin**
Closes spinnaker/spinnaker#969

### CODE REVIEW

**Comment by varikin** [comment]
@spinnaker/reviewers PTAL

**Comment by lwander** [comment]
LGTM

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/model/OpenstackSecurityGroup.groovy [hunk 1]
```diff
@@ -37,8 +42,64 @@ class OpenstackSecurityGroup implements SecurityGroup {
...
+
+  private static AddressableRange buildAddressableRangeFromCidr(String cidr) {
+    if (!cidr) {
+      return null
+    }
+
+    def rangeParts = cidr.split('/') as List
+
+    // If the cidr just a single IP address, use 32 as the mask
+    if (rangeParts.size() == 1) {
```
Comment (varikin) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/model/OpenstackSecurityGroup.groovy]: I had it elsewhere and moved it here as a static method. I thought I needed it in 2 locations, but I guess I only needed it in one place, but kept it as a static method anyways. I had pulled this logic from the Google security group caching.

Comment (lwander) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/model/OpenstackSecurityGroup.groovy]: Sounds good

**Pull request merged**